crates/ide/src/goto_type_definition.rs

```diff
@@ -55,19 +54,29 @@ pub(crate) fn goto_type_definition(
         Some((ty, node))
     })?;
 
-    let mut res = FxHashSet::default();
-    let mut workload = vec![ty.strip_references()];
-    while let Some(ty) = workload.pop() {
-        if let Some(adt) = ty.as_adt() {
-            res.insert(adt);
+    let mut res = Vec::new();
+    let mut push = |def: hir::ModuleDef| {
+        if let Some(nav) = def.try_to_nav(db) {
+            if !res.contains(&nav) {
```

Comment:
This is quadratic, but we hopefully won't see types large enough for it to matter.

Reply:
I doubt it would matter here, but we could store the module defs first and recollect over them to get the navs I suppose.